File: Modules/SMES/MES/SIE.Web.MES/ProjectDesigns/ChildCommands/BomCommands/TreeBomInitCommand.js
SIE.defineCommand("SIE.Web.MES.ProjectDesigns.ChildCommands.BomCommands.TreeBomInitCommand", {
    meta: { text: "引用标准Bom", group: "edit", iconCls: "icon-SectionExpandAll icon-blue" },
    canExecute: function (view) {
        var sel = view.getSelection();
        if (sel.length <= 0) {
            return false;
        }
        var flag = true;
        sel.forEach(i => {
            if (i.dirty) {
                flag = false;
                return;
            }
        });
        return flag;
    },
    execute: function (view) {
        var me = this;
        SIE.Msg.askQuestion("引用标准Bom会覆盖当前数据明细，是否继续？".t(), function () {
            me.initBom(view);
        });
        
    },
    initBom: function (view) {
        var data = {};
        SIE.Msg.wait("正在引用标准Bom...".t());
        view.execute({
            data: data,
            withIds: true,
            selectIds: view.getSelectionIds(),
            success: function (res) {
                if (res.Success) {
                    view.reloadData();
                    SIE.Msg.close();
                }
            }
        })
    }
})

File: Modules/SMES/MES/SIE.Web.MES/ProjectDesigns/ChildCommands/BomCommands/TreeBomInitCommand.cs
using SIE.MES.ProjectDesigns;
using SIE.MES.ProjectDesigns.ApiModels;
using SIE.Web.Command;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SIE.Web.MES.ProjectDesigns.ChildCommands.BomCommands
{
    /// <summary>
    /// 工艺资料产品Bom引用标准BOM命令
    /// </summary>
    public class TreeBomInitCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args"></param>
        /// <param name="scope"></param>
        /// <returns></returns>
        /// <exception cref="NotImplementedException"></exception>
        protected override object Excute(ViewArgs args, string scope)
        {
            var data = args.SelectedIds;
            RT.Service.Resolve<ProjectDesignController>().InitDesignTreeBoms(data);
            return true;
        }
    }
}
